File: R/app/src/main/java/com/example/voiceassistent/weatherapi/Forecast.java
package com.example.voiceassistent.weatherapi;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;
import java.util.List;

public class Forecast implements Serializable {

    @SerializedName("current")
    @Expose
    public Weather current;

    @SerializedName("location")
    @Expose
    public Region region;

    public class Weather{
        @SerializedName("temperature")
        @Expose
        public Integer temperature;

        @SerializedName("weather_descriptions")
        @Expose
        public List<String> weather_descriptions;
    }

    public class Region{
        @SerializedName("region")
        @Expose
        public String region;
    }
}

File: Rkt/app/src/main/java/com/example/voiceassistent/weatherapi/Forecast.kt
package com.example.voiceassistent.weatherapi

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.io.Serializable

class Forecast : Serializable {
    @JvmField
    @SerializedName("current")
    @Expose
    var current: Weather? = null

    @JvmField
    @SerializedName("location")
    @Expose
    var region: Region? = null

    inner class Weather {
        @JvmField
        @SerializedName("temperature")
        @Expose
        var temperature: Int? = null

        @JvmField
        @SerializedName("weather_descriptions")
        @Expose
        var weather_descriptions: List<String>? = null
    }

    inner class Region {
        @JvmField
        @SerializedName("region")
        @Expose
        var region: String? = null
    }
}